### DISCUSSION TITLE
How to get results after using the function runSimulationBatches

### DISCUSSION THREAD

**Discussion by teutonicod** [Open-Systems-Pharmacology/Forum#1554] [Questions & Problems] [answered]
Dear all,

I would like to use the `runSimulationBatches()` function in R. I think I got how to use the function, but I don't find how to get the plasma concentrations simulated by the batch simulation. I only get a sort of "code" as reported in the examples in the manual of the package.
Could you please help me on how to get the plasma concentrations generated by the simulation?

Below I reported the code I have so far.

``mySim <- loadSimulation(mySim.pkml)``
``mySimBatch <- createSimulationBatch(simulation = mySim , parametersOrPaths = myParamPath)``
``mySimBatch$addRunValues(parameterValues = XXXX)``
``simResults <- runSimulationBatches(mySimBatch)``

What I would need is how to go from here to the PK concentrations :-)
Since what I got after the run is something like this:
`[1] "14809397-c10f-4947-9f1b-64409124129b"`

Thank you very much in advance!

Donato

**Comment by PavelBal** [comment, marked as answer]
Hi Donato,

`simResults` is a named list, the names are composed from the id of the simulation batch and the id of the parameter values sets. The reason is that you can execute multiple `SimulationBatch` objects with multiple run values in parallel, and by knowing the ID you can then access the right results.

In your case, there is only one batch with one values set, so `simResults` is a list with only one entry. Just access the fist element of this list, which is a `SimulationResults` object. You can then access the outputs as described here https://www.open-systems-pharmacology.org/OSPSuite-R/dev/articles/run-simulation.html#running-individual-simulation-and-retrieving-the-results

**Comment by teutonicod** [comment]
Hi @PavelBal,

thanks a lot for your reply. It seems I managed to get the data from the simulation!
